### PR TITLE
Zombie burner ocassionally fires up napalm it's standing on

### PR DESCRIPTION
**Pull request by Night-Pryanik**
I feel that zombie burner just spamming napalm is kinda boring, so I added a chance to fire up the napalm it dropped - imagine it accidentally touches the puddle of napalm with the working pilot light of its flamethrower. Also made it fireproof so it won't catch up the fire it created.

### CODE REVIEW

**Comment by DracoGriffin** [comment]
This was one of the main concerns in #23729, so why feel the need to revert? And continues with the FIREPROOF so it will continue to generate firestorms.

Does the zombie even ever run out of napalm or is it basically just an infinite tank of napalm that somehow doesn't ever explode?

**Comment by ymber** [comment]
This needs limited napalm and unless its spawns get cut back to make it a top end cyberzombie the fireproofing is a problem. Heat and fire resistant gear is good but it isn't enough to let you stand in a lake of burning napalm indefinitely.

A mostly braindead zombie with a flamethrower would be a danger to itself and everything around it. With a 25% chance of setting napalm under it on fire it's going to be wandering around torching everything. That makes leaving it out of aggro range and waiting for it to kill itself a viable and likely optimal tactic to deal with them. I like the idea of a monster that's unstable and dangerous enough for the player to take advantage of, but I think this isn't a good implementation.

In #23729 there were a bunch of ideas proposed for biological blob evolution analogs for a flamethrower zombie and they would leave the design a lot more open for interesting variation and monster evolutions. The concepts where it works in tandem with other monsters are good because they keep it around things it can have interesting interactions with. The main point of the monster is that it's unstable and dangerous to whatever is near it and the player can capitalize on that or be damaged by it. Going down the synergy route and reworking it into a napalm boomer is the most obvious option, and given how it provides new and unique combat interactions and how many things you could do with an evolution tree like that I think it's the best option.

**Comment by Night-Pryanik** [comment]
> so why feel the need to revert?

This is not a revert, this is a different mechanics. The chance to generate fire is fairly small, and if it happens, the fire is likely to go out without any firestorms you afraid of.

> is it basically just an infinite tank of napalm?

The rate of napalm dripping is low, the tanks are rather big, so I think it is tolerable to consider them infinite.

**Comment by Night-Pryanik** [comment]
>in a lake of burning napalm

Zombie burner generates little puddles (250 ml, essentially a cup), not a lake.

> Going down the synergy route and reworking it into a napalm boomer is the most obvious option

Feel free to make it yourself if you like.

**Comment by Xhuis** [comment]
The flammability of items and terrain in Cataclysm is immensely high, as I believe that fire rework never got in. Unless the burner is standing in water or under rain, this will likely make any fires it starts - especially near or on furniture like counters - ignite and spiral out of control.

I'm not usually one to cite higher design, but what do you feel does this add to the game? It feels to me like difficulty for its own sake; this will only rarely serve as a threat to the player, and much more commonly, it will simply result in buildings randomly being on fire or destroyed when you roll into town. And, sure, burning buildings are interesting, but this is a strange way of doing it; it's an unintentional consequence of a random spawn using a random chance to light liquid on fire for the sake of doing it.

The concept of a flamethrower zombie is inherently dangerous and touchy due to how quickly fire spreads, as was evidenced in the original PR. It resulted in having to neuter it down to its current state to make it anything more than a zombie of mass destruction, and this change threatens to toss all that to the wind and do it anyway. :v

**Comment by ymber** [comment]
The fireproofing problem isn't that it can survive the leakage fires, it's that it can survive fine when it sets everything around it on fire.

The main problem though is that this enemy design misses most of the points that could make a fire monster interesting and different.

**Comment by thethunderhawk** [comment]
Oh I hadn’t realized that had been merged. @NightPryanik how do you feel about the “synergistic fire zed evolution tree” concept as a landing spot for this idea down the line? I agree that a fire zed is a good idea but do you think this is the best permanent implementation of that idea?

**Comment by Night-Pryanik** [comment]
> how do you feel about the “synergistic fire zed evolution tree” concept as a landing spot for this idea down the line? I agree that a fire zed is a good idea but do you think this is the best permanent implementation of that idea?

I like the idea of evolution tree, but now I don't have any good thoughts on what it should look like.

> Would you be open to a larger community brainstorm session on this? On the forums perhaps?

No, I'm already sick of discussion on reddit about my first autodoc PR.

**Comment by thethunderhawk** [comment]
>already sick of reddit discussion

Can’t blame you there. How about the forum? I think crowdsourcing some ideas might not be the worst.
Alternatively I could moderate the conversation myself and bring you any relevant ideas, that way you wouldn’t have to deal with the rabble.

**Comment by Night-Pryanik** [comment]
You are welcome to open the discussion on forum or reddit or somewhere else, if you like. 
But I don't give no guarantees that I will try to implement something.

**Comment by thethunderhawk** [comment]
>No guarantees that I’ll implement something

Of course. I just wanna get more ideas circulating. Cool if I post those here or should I open a suggestion?

**Comment by Night-Pryanik** [comment]
> should I open a suggestion?

This.

**Comment by thethunderhawk** [comment]
Sounds good, I’m opening a couple conversations after my workday today. Expect results in a couple days.

File: src/monmove.cpp [hunk 1]
```diff
@@ -1117,6 +1117,12 @@ bool monster::move_to( const tripoint &p, bool force, const float stagger_adjust
     if( has_flag( MF_DRIPS_NAPALM ) ) {
         if( one_in( 10 ) ) {
             g->m.add_item_or_charges( pos(), item( "napalm" ) );
+            if( one_in( 4 ) ) {
+                g->m.add_field( pos(), fd_fire, 2 );
+                add_msg( m_warning,
```
Comment (BevapDin) [inline src/monmove.cpp]: There needs to be a check that the player can actually see this before you print a message.

**Comment by kevingranade** [comment]
I don't think we want a firestarting zombie in general, maybe if the description were adjusted as outlined previously, but as it is now it feels way too arbitrary.

**Pull request closed without merging**